earthpy/spatial.py

```diff
@@ -30,14 +31,15 @@ def extent_to_json(ext_obj):
     """
 
     if type(ext_obj) == gpd.geodataframe.GeoDataFrame:
-        extent_json = mapping(box(*ext_obj.bounds.values[0]))
+        extent_json = mapping(box(*ext_obj.total_bounds))
     elif type(ext_obj) == list:
-        extent_json = mapping(box(ext_obj))
+        extent_json = mapping(box(*ext_obj))
     else:
-        raise ValueError("Please provide a geodataframe of a list of values - minx, miny, maxx, maxy")
+        raise ValueError("Please provide a geodataframe of a list of values.")
```

Comment:
Shortened this error message so that we only have to maintain the order of list elements in one place (the docstring)

Reply:
Also this avoids us having to deal with handling multiline exceptions

Reply:
ok @mbjoseph  there is a typo here (my typo) it should say 

`"Please provide a geodataframe or a list of values."`

Ok let's think about this. this is just testing that the object is of one of the correct types. And i suppose if the min and max values are mixed up that is pretty hard to test for... so this makes sense! Just fix that one small typo pretty please but very good update here. i appreciate it!

Reply:
Typo is fixed with 4bcaca912f2f61a04547e83525d81db63f2e45e4 - I would recommend in the future that we try to use duck typing rather than testing object types unless absolutely necessary, to get extra cool "Pythonic" street cred.

Reply:
sure but for this particular instance, can you show me how duck typing would be applied? this is not a class, it's logic to determine how to process the output. I do love me some python street cred tho!

Reply:
Word! I think in this case it's fine to have it as written, but there is a slight loss of generality. For instance, if a user wanted to pass in a tuple of xmin, ymin, xmax, ymax, there's not much reason to raise an error assuming that those can be unpacked in the `box` function, and that we can check that `xmin <= xmax` and `ymin <= ymax`.